addons/stock/tests/test_move.py

```diff
@@ -4186,16 +4186,14 @@ def test_put_in_pack_1(self):
         picking.action_confirm()
         picking.action_assign()
         self.assertEqual(self.env['stock.quant']._get_available_quantity(self.product, self.stock_location), 0)
-        move1.quantity_done = 1
         picking.put_in_pack()
+        self.assertEqual(len(picking.move_line_ids), 1)
+        self.env['stock.quant']._update_available_quantity(self.product, self.stock_location, 1)
+        picking.action_assign()
+        self.assertEqual(len(picking.move_line_ids), 1)
         self.assertEqual(self.env['stock.quant']._get_available_quantity(self.product, self.stock_location), 0)
-        self.assertEqual(len(picking.move_line_ids), 2)
-        unpacked_ml = picking.move_line_ids.filtered(lambda ml: not ml.result_package_id)
-        self.assertEqual(unpacked_ml.product_qty, 1)
-        unpacked_ml.qty_done = 1
-        picking.put_in_pack()
-        self.assertEqual(len(picking.move_line_ids), 2)
-        self.assertEqual(self.env['stock.quant']._get_available_quantity(self.product, self.stock_location), 0)
-        picking.button_validate()
-        self.assertEqual(self.env['stock.quant']._get_available_quantity(self.product, self.stock_location), 0)
+        res_dict = picking.button_validate()
+        self.assertEqual(res_dict.get('res_model'), 'stock.immediate.transfer')
+        wizard = self.env[(res_dict.get('res_model'))].browse(res_dict.get('res_id'))
+        wizard.process()
```

Comment:
keep this old test and add a new one, here you changed what was tested and i don't see why you would need an immediate transfer anywhere?

test1)
one move of product 1 reserved
one move of product 2 reserved 

hit put in pack
you have two move line qty done 1 result package package1

test2)
one move of product 1 reserved
one move of product 2 reserved 
product 1 set qty done 1
put in pack
product 2 set qty done 2
put in pack
you have two move line qty done 1 result package package1 and package2

that would be a start

Reply:
Done